File: Domain/Notification.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Domain
{
    public class Notification
    {
        public Guid Id { get; set; }
        
        [Required]
        public string RecipientId { get; set; }
        public AppUser Recipient { get; set; }
        public NotificationType Type { get; set; }
        
        [Required]
        public string Message { get; set; }
        public bool IsRead { get; set; } = false;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? ReadAt { get; set; }
        
        // Optional relationships for context
        public Guid? CommentId { get; set; }
        public Comment Comment { get; set; }
        
        public Guid? TicketId { get; set; }
        public Ticket Ticket { get; set; }
    }

    public enum NotificationType
    {
        Mention,
        CommentReply,
        TicketAssigned,
        TicketStatusChanged
    }
}

File: client-app/src/app/models/notification.ts
export interface Notification {
  id: string;
  recipientId: string;
  type: NotificationType;
  message: string;
  isRead: boolean;
  createdAt: string;
  readAt: string | null;
  commentId: string | null;
  ticketId: string | null;
  ticketTitle: string | null;
  authorDisplayName: string | null;
  authorUsername: string | null;
}

export enum NotificationType {
  Mention = 0,
  CommentReply = 1,
  TicketAssigned = 2,
  TicketStatusChanged = 3,
}
